src/audio/buffers/ring_buffer.c

```diff
@@ -239,6 +239,21 @@ static int ring_buffer_release_data(struct sof_source *source, size_t free_size)
 	return 0;
 }
 
+int ring_buffer_module_unbind(struct sof_sink *sink)
+{
+	struct ring_buffer *ring_buffer = ring_buffer_from_sink(sink);
+
+	CORE_CHECK_STRUCT(&ring_buffer->audio_buffer);
+
+	/* in case of disconnection, invalidate all cache. This method is guaranteed be called on
+	 * core that have been using sink API
+	 */
+	ring_buffer_invalidate_shared(ring_buffer, ring_buffer->_data_buffer,
```

Comment:
the name of this function is actually somewhat confusing: usually in SOF "shared" means "uncached"